apps/settings/js/panels/operator_settings/model/auto_selection_model.js

```diff
@@ -0,0 +1,82 @@
+define(function() {
+  'use strict';
+
+  var Defer = require('modules/defer');
+  var FeatureModel = require('modules/feature_model');
+  var MobileConnectionWrapper =
+    require('panels/operator_settings/mobile_connection_wrapper');
+
+  // Auto selection state
+  const AS_STATE = {
+    UNKNOWN: 0,
+    ENABLED: 1,
+    DISABLED: 2
+  };
+
+  var AutoSelectionModel = function(connWrapper) {
+    return FeatureModel({
+      onInit: function(onchange) {
+        this._pendingState = null;
+        this._defer = null;
+        connWrapper.observe('state', (newState) => {
+          this._checkPendingState(newState);
+        });
+        return AS_STATE.ENABLED;
```

Comment:
Not sure this default value is right or not, if users set the preference to **manual** before, AutoSelection is `enabled` here ?